### ISSUE TITLE
New-AzureRmAutoscaleRule not setting Direction

### ISSUE THREAD

**Issue by MarkDordoy** [Azure/azure-powershell#7267]
### Description
This issue has been visible to me within the AzureRM 6.8.1 update

Running the cmdlet `New-AzureRMAutoscaleRule` fails to set the scale direction. Both Increase and Decrease seem to set the object property to `None`
### Script/Steps for Reproduction

```powershell

$scaleUp = New-AzureRmAutoscaleRule -MetricName 'CpuPercentage' -MetricResourceId $webApp.Id -Operator GreaterThan -MetricStatistic Average -Threshold 70 -TimeGrain 00:01:00 -ScaleActionCooldown 00:10:00 -ScaleActionDirection Increase -ScaleActionScaleType ChangeCount -ScaleActionValue '1' 

$scaleUp.ScaleAction

       Type Direction Value Cooldown
       ---- --------- ----- --------
ChangeCount      None 1     00:10:00
```

### Module Version

5.1.3      AzureRM.Insights
5.5.1      AzureRM.Profile
5.0.9      AzureRM.Websites
6.8.1      AzureRM

### Environment Data

```powershell
$PSVersionTable
Name                           Value
----                           -----
PSVersion                      5.1.17134.228
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.17134.228
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1

```

### Debug Output

```
DEBUG: 09:42:26 - NewAzureRmAutoscaleRuleCommand begin processing with ParameterSet '__AllParameterSets'.
DEBUG: 09:42:26 - using account id 'mdosys@clarksons.co.uk'...
DEBUG: AzureQoSEvent: CommandName - New-AzureRmAutoscaleRule; IsSuccess - True; Duration - 00:00:00.0030414; Exception - ;
DEBUG: Finish sending metric.
DEBUG: 09:42:26 - NewAzureRmAutoscaleRuleCommand end processing.
DEBUG: 09:42:26 - NewAzureRmAutoscaleRuleCommand end processing.
```

**Comment by cormacpayne** [comment]
@gucalder Hey Guillermo, would you mind taking a look at this issue?

**Comment by sqlrohit** [comment]
I'm also facing the similar issue with both
New-AzureRmAutoscaleRule
New-AzureRmAutoscaleProfile
I've tried to explain this issue in below post
https://social.microsoft.com/Forums/en-US/2460a419-813c-4749-b86c-75cd883c7671/creating-scale-out-rules-for-app-service-plans-using-powershell?forum=AzureAPIApps

In order to understand the issue try below 
--------------
#Create Rule1
$AutoScaleRule1 = New-AzureRmAutoscaleRule -MetricName 'CpuPercentage' -MetricResourceId "/subscriptions/$subid/resourceGroups/$RsgName/providers/Microsoft.Web/serverFarms/$CurrASP" -MetricStatistic Average -Operator GreaterThanOrEqual -Threshold 80 -TimeGrain 00:01:00 -ScaleActionDirection Increase -ScaleActionScaleType ChangeCount -ScaleActionValue 1 -ScaleActionCooldown 00:05:00 -TimeWindow 00:10:00 -TimeAggregationOperator Average

#Print Values and notice it's not updating Direction
$AutoScaleRule1.ScaleAction
$AutoScaleRule1.MetricTrigger

#Manually Update value for Direction & OperatorProperty
$AutoScaleRule1.ScaleAction.Direction='Increase'
$AutoScaleRule1.MetricTrigger.OperatorProperty='GreaterThanOrEqual'

#Creating Autoscale Profile With Above Rule
$AutoScaleProfile = New-AzureRmAutoscaleProfile -Name "Default" -DefaultCapacity "1" -MaximumCapacity "6"  -MinimumCapacity "1" -Rule $AutoScaleRule1

#Print Values 
$AutoScaleProfile.Rules[0].MetricTrigger
$AutoScaleProfile.Rules[0].ScaleAction

#You've to again manually update values for OperatorProperty & Direction as it's not updated properly
$AutoScaleProfile.Rules[0].MetricTrigger.OperatorProperty="GreaterThanOrEqual"
$AutoScaleProfile.Rules[0].ScaleAction.Direction="Increase"
--------------

#Creating the AutoScale Setting
-------------
Add-AzureRmAutoscaleSetting -Location 'West Europe' -Name $SettingName -ResourceGroupName $RsgName -TargetResourceId "/subscriptions/$subid/resourceGroups/$RsgName/providers/microsoft.web/serverFarms/$CurrASP" -AutoscaleProfile $AutoScaleProfile
-------------

Please let me know.

**Comment by Falconwmua** [comment]
This is also occurring in version 5.1.4. The system accepts the rule as is and defaults to Decrease.

**Comment by wgv-srsedate** [comment]
We are also having this issue, and it is still present in **AzureRM 6.10.0**.  Even if you explicitly set the enum value for direction in your ScaleRule object, it still defaults to **None** when sent to Azure.  This translates to a **Decrease** behavior.

There's no programmatic workaround that I can see without downgrading your AzureRM module, and I'm not even sure what version the issue showed up in.

**Comment by wgv-srsedate** [comment]
This issue is verified fixed as of 6.12, at least for our purposes.
